core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -558,7 +565,7 @@ sealed trait ZIO[-R, +E, +A] extends Serializable { self =>
    * function. This can be used to lift a "smaller" error into a "larger"
    * error.
    */
-  final def mapError[E2](f: E => E2): ZIO[R, E2, A] =
+  final def mapError[E2](f: E => E2)(implicit ev: CanFail[E]): ZIO[R, E2, A] =
```

Comment:
This one is useful since there is no need to map the error when `E` = Nothing, since it's already a subtype of all other errors.